quinn/src/connection.rs

```diff
@@ -29,15 +29,57 @@ pub use crate::builders::{
 use crate::{ConnectionEvent, EndpointEvent};
 
 /// Connecting future
-pub struct ConnectingFuture(Option<ConnectionDriver>);
+pub struct Connecting(Option<ConnectionDriver>);
 
-impl ConnectingFuture {
+impl Connecting {
     pub(crate) fn new(conn: ConnectionRef) -> Self {
         Self(Some(ConnectionDriver(conn)))
     }
+
+    /// Convert into a 0-RTT or 0.5-RTT connection at the cost of weakened security
+    ///
+    /// Additionally returns a future that will yield the connection's [`IncomingStreams`] when the
+    /// connection is established.
```

Comment:
This part seems to be missing. I'm not wedded to the specific approach--it was just the easy thing to do--but as is there's no way for a 0-RTT application to determine when the handshake has completed. Although, now that I think about it, I'm not actually certain whether that's really necessary...

Reply:
Yeah, I did remove it somewhat consciously -- it didn't seem necessary, and this approach is simpler and provides a nice symmetry between the `<Connecting as Future>::Item` and the `into_0rtt()` `Ok` value. I'll update the documentation here and let me know if you think it needs to be a `Future` after all.

Reply:
I'm happy to wait and see if anyone comes up with a reason to want it. Only thing I can think of is a very specific case of status feedback.